### ISSUE TITLE
[Bug]: page.pdf() is much slower after upgrading from chromium 117 to 119

### ISSUE THREAD

**Issue by majdnaji** [puppeteer/puppeteer#11494]
### Minimal, reproducible example

```TypeScript
const browser = await puppeteer.launch({
    executablePath: "/usr/bin/chromium-browser",
    headless: "new",
    args: [
      "--disable-features=IsolateOrigins",
      "--disable-site-isolation-trials",
      "--autoplay-policy=user-gesture-required",
      "--disable-background-networking",
      "--disable-background-timer-throttling",
      "--disable-backgrounding-occluded-windows",
      "--disable-breakpad",
      "--disable-client-side-phishing-detection",
      "--disable-component-update",
      "--disable-default-apps",
      "--disable-dev-shm-usage",
      "--disable-domain-reliability",
      "--disable-extensions",
      "--disable-features=AudioServiceOutOfProcess",
      "--disable-hang-monitor",
      "--disable-ipc-flooding-protection",
      "--disable-notifications",
      "--disable-offer-store-unmasked-wallet-cards",
      "--disable-popup-blocking",
      "--disable-print-preview",
      "--disable-prompt-on-repost",
      "--disable-renderer-backgrounding",
      "--disable-setuid-sandbox",
      "--disable-speech-api",
      "--disable-sync",
      "--hide-scrollbars",
      "--ignore-gpu-blacklist",
      "--metrics-recording-only",
      "--mute-audio",
      "--no-default-browser-check",
      "--no-first-run",
      "--no-pings",
      "--no-sandbox",
      "--no-zygote",
      "--password-store=basic",
      "--use-gl=swiftshader",
      "--use-mock-keychain",
    ],
    timeout: 300000,
  });;
  console.timeLog("generate", `browser launched ${name}`);
  try {
    const page = await browser.newPage();
    console.timeLog("generate", `page created ${name}`);

    await page.setContent(`<div>hi</div>`, { waitUntil: "networkidle0" });
    console.timeLog("generate", `content set ${name}`);
    // Download the PDF
    const pdfPage = await page.pdf({
      path: `${name}.pdf`,
      displayHeaderFooter,
      footerTemplate,
      headerTemplate,
      format: landscape ? "a4" : format,
      height,
      width,
      landscape,
      margin,
      omitBackground,
      pageRanges,
      preferCSSPageSize,
      printBackground,
      scale,
      timeout,
    });
    console.timeLog("generate", `pdf created ${name}`);
    return pdfPage;
  } catch (err) {
    return Promise.reject(err);
  } finally {
    await browser.close();
    console.timeLog("generate", `browser closed ${name}`);
    console.timeEnd("generate");
  }
```

### Error string

no error

### Bug behavior

- [ ] Flaky
- [ ] PDF

### Background

i have a cloud run that is an api for generating PDFs based on html input, i'm using node:16-alpine, puppeteer, chromium and it was running fine until alpine upgraded it's chromium package from 117.0.5938.62-r0 to 119.0.6045.159-r0 and execution time of the pdf function tripled. i thought about setting chromium version to fixed in my Dockerfile but it's not allowed on alpine. 
any help?

### Expectation

the performance shouldn't have changed

### Reality

execution time tripled

### Puppeteer version

21.1.1

### Node version

16.20.2

### Package manager

npm

### Package manager version

10.1.0

### Operating system

Linux

**Comment by Lightning00Blade** [comment]
Unable to reproduce locally with the browser Puppeteer installs. 
Note: I had to remove some of the options from the PDF generation as you have not provided the value for them 
Can you try using the Browser that Puppeteer provides, you can see the list of version to Puppeteer version here - https://pptr.dev/chromium-support ? 


![image](https://github.com/puppeteer/puppeteer/assets/34244704/51626193-66ff-42d3-ade3-41c872485ad2)

**Comment by majdnaji** [comment]
> Unable to reproduce locally with the browser Puppeteer installs. Note: I had to remove some of the options from the PDF generation as you have not provided the value for them. Can you try using the Browser that Puppeteer provides and see if the issue is there as well, you can see the list of Chrome version to Puppeteer version here - https://pptr.dev/chromium-support ?
> 
> ![image](https://private-user-images.githubusercontent.com/34244704/287995362-51626193-66ff-42d3-ade3-41c872485ad2.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDE3NzEzMDIsIm5iZiI6MTcwMTc3MTAwMiwicGF0aCI6Ii8zNDI0NDcwNC8yODc5OTUzNjItNTE2MjYxOTMtNjZmZi00MmQzLWFkZTMtNDFjODcyNDg1YWQyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMDUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjA1VDEwMTAwMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWNjMWRiNjQzZWIzNmM0MzkwNDU2Njg0YTliNGJmODE3NWFmNTk4MGI0M2NlZTNmZGYzYWM1MGVhY2Q2Y2FmYzkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.3xwSFA5fkJfNhKaA5kOkEEKYerijLAOefUMrEda658s)

i'm testing with larger HTML content, multiple pages with high res images, on a smaller input the difference won't be as noticable

**Comment by Lightning00Blade** [comment]
Tried on the HTML spec single page site [https://html.spec.whatwg.org/](https://html.spec.whatwg.org/) and it seem its faster on the new version, mind you this is only text with links.
Could you try to hide the images from the page to see if that is the cause of the slow down? 
Also what happens if you only use the `path: `${name}.pdf`,` in the options for `page.pdf`,  does the issue reproduce there?

![image](https://github.com/puppeteer/puppeteer/assets/34244704/191b54b7-a634-4a54-b06c-a08a7568ed63)

**Comment by majdMediaslide** [comment]
> Tried on the HTML spec single page site https://html.spec.whatwg.org/ and it seem its faster on the new version, mind you this is only text with links. Could you try to hide the images from the page to see if that is the cause of the slow down? Also what happens if you only use the `path: ${name}.pdf,` in the options for `page.pdf`, does the issue reproduce then?
> 
> ![image](https://private-user-images.githubusercontent.com/34244704/288073550-191b54b7-a634-4a54-b06c-a08a7568ed63.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDE5NTM3MjgsIm5iZiI6MTcwMTk1MzQyOCwicGF0aCI6Ii8zNDI0NDcwNC8yODgwNzM1NTAtMTkxYjU0YjctYTYzNC00YTU0LWIwNmMtYTA4YTc1NjhlZDYzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMDclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjA3VDEyNTAyOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWViMmIwOTM1ZDlkNDRhODU4ZTIxNGVjNDRiOTllMDdmYmIyZDA5ZTFkN2MwNDcwOWRkMGQ1NWM0ZWU1ZDhhZTUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.Z5HO-V0H-Fds7tkfyZ38x4OGSaDIhhKvKKyw4YuPQ38)

without images it's working fine
i removed all passed options on the 119 version except `printBackground:true`  and still the difference is too big
119 request stats
![image](https://github.com/puppeteer/puppeteer/assets/151499562/263dc06d-4fdc-450f-a2e4-1658599bb84f)
117 version stats
![image](https://github.com/puppeteer/puppeteer/assets/151499562/b196739d-f46e-427a-b1f1-4dfa52d0290f)

**Comment by Lightning00Blade** [comment]
Yeah, I can see the issue if I have image elements (~50) that are sort of big - 

For Puppeteer v21.4.0 - `Chrome/118.0.5993.70` - `7.628s`
For Puppeteer v21.6.0 - `Chrome/119.0.6045.105` - `1:09.572 (m:ss.mmm)` (`headless: "new"`)
For Puppeteer v21.6.0 - `Chrome/119.0.6045.105` - `1:10.565 (m:ss.mmm)` (`headless: false`)

I will open a bug about it on the Chromium repo and link it back on this comment.
